src/Features/CSharp/Portable/CodeFixes/HideBase/HideBaseCodeFixProvider.AddNewKeywordAction.cs

```diff
@@ -25,16 +33,36 @@ protected override async Task<Document> GetChangedDocumentAsync(CancellationToke
             {
                 var root = await _document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-                var newNode = GetNewNode(_node);
+                var newNode = await GetNewNode(_node, cancellationToken).ConfigureAwait(false);
                 var newRoot = root.ReplaceNode(_node, newNode);
 
                 return _document.WithSyntaxRoot(newRoot);
             }
 
-            private SyntaxNode GetNewNode(SyntaxNode node)
+            private async Task<SyntaxNode> GetNewNode(SyntaxNode node, CancellationToken cancellationToken)
             {
                 var generator = SyntaxGenerator.GetGenerator(_document);
-                return generator.WithModifiers(node, generator.GetModifiers(node).WithIsNew(true));
+                var newNode = generator.WithModifiers(node, generator.GetModifiers(node).WithIsNew(true));
+
+                var options = await _document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+                var option = options.GetOption(PreferredModifierOrderOption);
+                if (!OrderModifiersHelper.TryGetOrComputePreferredOrder(option.Value, out var preferredOrder))
```

Comment:
❔ Can we expand this condition to include the case where the modifiers are not ordered at the start of this method? This would prevent the code fix from reordering unrelated modifiers.